src/colvarbias_abf.h

```diff
@@ -137,11 +128,29 @@ class colvarbias_abf : public colvarbias {
   // Share between replicas -- may be called independently of update
   virtual int replica_share();
 
+  // Share data needed for CZAR between replicas - called before output only
+  int replica_share_CZAR();
+
+  /// Report the frequency at which this bias needs to communicate with replicas
   virtual size_t replica_share_freq() const;
 
-  // Store the last set for shared ABF
-  colvar_grid_gradient  *last_gradients;
-  colvar_grid_count     *last_samples;
+  // Data just after the last share (start of cycle) in shared ABF
+  std::unique_ptr<colvar_grid_gradient> last_gradients;
+  std::unique_ptr<colvar_grid_count>    last_samples;
+  // eABF/CZAR local data last shared
+  std::unique_ptr<colvar_grid_gradient> last_z_gradients;
+  std::unique_ptr<colvar_grid_count>    last_z_samples;
+  // ABF data from local replica only in shared ABF
+  std::unique_ptr<colvar_grid_gradient> local_gradients;
+  std::unique_ptr<colvar_grid_count>    local_samples;
+  std::unique_ptr<integrate_potential>  local_pmf;
+  // eABF/CZAR data collected from all replicas in shared eABF on replica 0
+  // if non-shared, aliases of regular CZAR grids, for output purposes
+  colvar_grid_gradient *global_z_gradients;
```

Comment:
I see the lines above use smart pointers like `std::unique_ptr`, but here the raw pointers are still used. Could they be shared pointers?